QuizApp/QuizApp/UI/Screens/User/UserInfoView.swift

```diff
@@ -0,0 +1,71 @@
+import SwiftUI
+
+struct UserInfoView: View {
+
+    let onLogoutTap: () -> Void
+    let onNameChange: () -> Void
+
+    @ObservedObject var userInfo: UserInfo
+
+    init(userInfo: UserInfo, onLogoutTap: @escaping () -> Void = {}, onNameChange: @escaping () -> Void = {}) {
+        self.userInfo = userInfo
+        self.onLogoutTap = onLogoutTap
+        self.onNameChange = onNameChange
+    }
+
+    var body: some View {
+        HStack {
+            VStack(alignment: .leading) {
+                Text("USERNAME")
+                    .font(.subtitle4)
+                    .foregroundColor(.white)
+                    .padding(.top, 40)
+
+                TextField("", text: $userInfo.username)
+                    .font(.heading4)
+                    .foregroundColor(.white)
+                    .disabled(true)
+
+                Text("NAME")
+                    .font(.subtitle4)
+                    .foregroundColor(.white)
+                    .padding(.top, 40)
+
+                TextField("", text: $userInfo.name, onCommit: { onNameChange() })
```

Comment:
This is deprecated: https://developer.apple.com/documentation/swiftui/textfield/init(_:text:oncommit:)-1zeco?changes=latest_minor
Read there how to change it.